test/testing/test_services.py

```diff
@@ -31,6 +35,23 @@ def reset_mock():
     handle_event.reset_mock()
 
 
+@pytest.yield_fixture
+def spawn_thread():
```

Comment:
maybe a docstring explaining that this makes sure any greenlets are killed at the end of the test